File: spring-boot-backend/src/main/java/de/maschmi/blog/backend/mapper/StorageBoxMapper.java
package de.maschmi.blog.backend.mapper;

import de.maschmi.blog.backend.poc.RoomData;
import de.maschmi.blog.backend.poc.StorageBoxData;
import de.maschmi.blog.backend.poc.ValueClassMapper;
import domain.Room;
import domain.StorageBox;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(uses = ValueClassMapper.class)
public interface StorageBoxMapper {

    StorageBoxData storageBoxToData(StorageBox box);
    StorageBox storageBoxDataToDomain(StorageBoxData data);

    @Mapping(target = "copy", ignore = true)
    RoomData roomToData(Room room);
    @Mapping(target = "copy", ignore = true)
    Room roomDataToDomain(RoomData data);
}

File: spring-boot-backend/src/main/kotlin/de/maschmi/blog/backend/poc/StorageBoxMapper.kt
package de.maschmi.blog.backend.poc

import domain.Room
import domain.RoomName
import domain.StorageBox
import org.mapstruct.Mapper

@Mapper(uses = [ValueClassMapper::class])
interface StorageBoxMapper {

    fun storageBoxTodata(box: StorageBox): StorageBoxData
    fun storageBoxDataToDomain(data: StorageBoxData): StorageBox

    fun roomTodata(room: Room): RoomData
    fun roomDataToDomain(data: RoomData): Room

    fun toRoomName(value: String): RoomName = RoomName(value = value)
    fun toRoomNameData(value: RoomName): String = value.value

    fun roomToString(room: Room): String = room.name.value
    fun stringToRoom(value: RoomData): Room = Room(RoomName(value.room))
}